2_adventures/001/Randhir123/05/lib/05/fibonacci.rb

```diff
@@ -0,0 +1,29 @@
+class Fibonacci
+	
+	def initialize()
+		@cache = {}	
+	end
+	
+	def get(n)
+		if n == 0 || n == 1
+			return n
+		else
+			return get(n-2) + get(n-1)
+		end
```

Comment:
Of course, like Pat says... readability.  But considering there is something to be said when you see a return, it brings your attention to "I can return from here and from here" wherever you see that return. In this case, it is a miscommunication in a way, because really the return is the line 12 where the `end` is.  You aren't quite letting the if statement finish.  You are interrupting your conditional statement with a premature exit from that statement.